src/runmode-dpdk.c

```diff
@@ -355,8 +356,32 @@ static void ConfigSetIface(DPDKIfaceConfig *iconf, const char *entry_str)
 static int ConfigSetThreads(DPDKIfaceConfig *iconf, const char *entry_str)
 {
     SCEnter();
-    const char *active_runmode = RunmodeGetActive();
+    static int32_t remaining_auto_cpus = -1;
+    static uint32_t total_cpus = 0;
+    if (!threading_set_cpu_affinity) {
+        SCLogError("DPDK runmode requires configured thread affinity");
+        SCReturnInt(-EINVAL);
+    }
+
+    ThreadsAffinityType *wtaf = GetAffinityTypeFromName("worker-cpu-set");
+    uint32_t sched_cpus = UtilAffinityGetAffinedCPUNum(wtaf);
+    ThreadsAffinityType *mtaf = GetAffinityTypeFromName("management-cpu-set");
+    if (sched_cpus == UtilCpuGetNumProcessorsOnline()) {
+        SCLogWarning(
+                "\"all\" specified in worker CPU cores affinity, excluding management threads");
+        UtilAffinityCpusExclude(wtaf, mtaf);
+        sched_cpus = UtilAffinityGetAffinedCPUNum(wtaf);
+    }
 
+    if (sched_cpus == 0) {
+        SCLogError("No worker CPU cores with configured affinity were configured");
+        SCReturnInt(-EINVAL);
+    } else if (UtilAffinityCpusOverlap(wtaf, mtaf) != 0) {
+        SCLogWarning("Worker threads should not overlap with management threads in the CPU core "
```

Comment:
any reason to have this as a warning, not an error?

Reply:
Didn't want to limit users too much in case there would be a use-case that they want to configure.